TheMovieDb/Modules/Home/HomeRemoteDataManager.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  HomeRemoteDataManager.swift
+//  TheMovieDb
+//
+//  Created by Javier Cueto on 14/11/21.
+//  
+//
+
+import Foundation
+
+final class HomeRemoteDataManager: HomeRemoteDataManagerInputProtocol {
+    
+    var remoteRequestHandler: HomeRemoteDataManagerOutputProtocol?
+    private let group = DispatchGroup()
+    private var movies: [MovieGroupSections: [Movie]] = [:]
+    
+    func fetchMovies() {
+        fetchData(typeMovieSection: .popular)
+        fetchData(typeMovieSection: .upcoming)
+        fetchData(typeMovieSection: .topRated)
+        fetchData(typeMovieSection: .playingNow)
+        fetchData(typeMovieSection: .trending)
```

Comment:
enum can be iterable and you can even get an array of enums if use `CaseIterable` this can help you to reduce this code into a map or a foreach

Reply:
I´ve fixed with this line of code  // MovieGroupSections.allCases.forEach { fetchData(typeMovieSection: $0) }